File: app/src/main/java/com/marinov/openfei/LoginLogic.kt
package com.marinov.openfei

import android.content.Context
import android.webkit.CookieManager
import androidx.core.content.edit
import kotlinx.coroutines.Dispatchers
import kotlinx.coroutines.withContext
import org.jsoup.Connection
import org.jsoup.Jsoup

data class LoginResult(
    val success: Boolean,
    val errorMessage: String = ""
)

object LoginLogic {
    const val LOGIN_URL = "https://interage.fei.org.br/secureserver/portal"
    suspend fun performLogin(user: String, pass: String, context: Context): LoginResult = withContext(Dispatchers.IO) {
        try {
            // 1. Acessar a página de login para pegar os cookies iniciais e o Token
            val resGet = Jsoup.connect(LOGIN_URL)
                .method(Connection.Method.GET)
                .execute()

            val docGet = resGet.parse()
            val token = docGet.select("input[name=__RequestVerificationToken]").`val`()
            val cookies = resGet.cookies()

            if (token.isEmpty()) {
                return@withContext LoginResult(false, "Não foi possível obter o token de segurança da página.")
            }

            // 2. Fazer o POST com os dados de login
            val resPost = Jsoup.connect("$LOGIN_URL/")
                .data("__RequestVerificationToken", token)
                .data("Usuario", user)
                .data("Senha", pass)
                .cookies(cookies)
                .method(Connection.Method.POST)
                .followRedirects(true)
                .execute()

            val docPost = resPost.parse()
            val isSuccess = docPost.select("#btn-login").isEmpty() // Se o botão sumiu, logou com sucesso

            if (isSuccess) {
                // 3. Salvar os cookies de sessão no CookieManager com expiração de 15 minutos
                val cookieManager = CookieManager.getInstance()
                cookieManager.setAcceptCookie(true)
                resPost.cookies().forEach { (key, value) ->
                    cookieManager.setCookie(
                        "https://interage.fei.org.br",
                        "$key=$value; Domain=interage.fei.org.br; Path=/; Max-Age=900"
                    )
                }
                cookieManager.flush()

                // 4. Atualizar estado de login nas SharedPreferences
                context.getSharedPreferences(LoginActivity.PREFS_LOGIN, Context.MODE_PRIVATE).edit {
                    putBoolean(LoginActivity.KEY_IS_LOGGED_IN, true)
                    putString(LoginActivity.KEY_USER, user)
                    putString(LoginActivity.KEY_PASS, pass)
                }

                return@withContext LoginResult(true)
            } else {
                // Falha no login → garantir que estado fique como false
                context.getSharedPreferences(LoginActivity.PREFS_LOGIN, Context.MODE_PRIVATE).edit {
                    putBoolean(LoginActivity.KEY_IS_LOGGED_IN, false)
                }
                val errorMsg = docPost.select(".field-validation-error").text()
                return@withContext LoginResult(false, errorMsg.ifEmpty { "Usuário ou senha incorretos." })
            }
        } catch (e: Exception) {
            // Em caso de erro, também marcamos como não logado
            context.getSharedPreferences(LoginActivity.PREFS_LOGIN, Context.MODE_PRIVATE).edit {
                putBoolean(LoginActivity.KEY_IS_LOGGED_IN, false)
            }
            return@withContext LoginResult(false, "Erro de conexão: ${e.message}")
        }
    }
}

File: Linux/login_logic.py
import aiohttp
from bs4 import BeautifulSoup
from dataclasses import dataclass


@dataclass
class LoginResult:
    success: bool
    error_message: str = ""


class LoginLogic:
    LOGIN_URL = "https://interage.fei.org.br/secureserver/portal"
    USER_AGENT = (
        "Mozilla/5.0 (Linux; Android 16; sdk_gphone64_x86_64 Build/BE2A.250530.026.D1; wv) "
        "AppleWebKit/537.36 (KHTML, like Gecko) Version/4.0 Chrome/133.0.6943.137 Mobile Safari/537.36"
    )

    @staticmethod
    async def perform_login(user: str, password: str, session: aiohttp.ClientSession) -> LoginResult:
        headers = {'User-Agent': LoginLogic.USER_AGENT}
        try:
            async with session.get(LoginLogic.LOGIN_URL, headers=headers) as resp:
                if resp.status != 200:
                    return LoginResult(False, f"Erro ao acessar página de login (HTTP {resp.status})")
                html = await resp.text()

            soup = BeautifulSoup(html, 'html.parser')
            token_input = soup.select_one('input[name="__RequestVerificationToken"]')
            if not token_input or not token_input.get('value'):
                return LoginResult(False, "Token de segurança não encontrado na página.")
            token = token_input['value']

            data = {
                '__RequestVerificationToken': token,
                'Usuario': user,
                'Senha': password
            }
            async with session.post(LoginLogic.LOGIN_URL + '/', data=data, headers=headers) as resp:
                if resp.status != 200:
                    return LoginResult(False, f"Erro no envio do login (HTTP {resp.status})")
                html = await resp.text()

            soup = BeautifulSoup(html, 'html.parser')
            if soup.select_one('#btn-login') is None:
                return LoginResult(success=True)
            else:
                error_elem = soup.select_one('.field-validation-error')
                error_msg = error_elem.text.strip() if error_elem else "Usuário ou senha incorretos."
                return LoginResult(success=False, error_message=error_msg)

        except aiohttp.ClientError as e:
            return LoginResult(False, f"Erro de conexão: {str(e)}")
        except Exception as e:
            return LoginResult(False, f"Erro inesperado: {str(e)}")